platforms/vita/dirent.h

```diff
@@ -26,50 +26,54 @@
 
 struct DIR
 {
+	DIR() : is_first(true) {}
+
 	bool is_first;
 	SceUID descriptor;
 };
 
 struct dirent
 {
-	char d_name[_MAX_PATH];
+	dirent() : buffer(_MAX_PATH, '\0'), d_name(const_cast<char*>(buffer.data())) {}
+
+	std::string buffer;
+	const char *d_name;
 };
 
 inline DIR* opendir(const char *filename)
 {
-	DIR *dirp = (DIR*) malloc(sizeof(DIR));
-	dirp->is_first = 1;
-	dirp->descriptor = sceIoDopen(filename);
+	auto *dir = new DIR{};
+	dir->descriptor = sceIoDopen(filename);
 
-	if (dirp->descriptor <= 0) {
-		free(dirp);
-		return NULL;
+	if (dir->descriptor <= 0) {
+		delete dir;
+		return nullptr;
 	}
 
-	return dirp;
+	return dir;
 }
 
-inline dirent* readdir(DIR *dirp)
+inline dirent* readdir(DIR *dir)
 {
 	static dirent de;
 	//vitasdk kind of skips current directory entry..
-	if (dirp->is_first) {
-		dirp->is_first = 0;
+	if (dir->is_first) {
+		dir->is_first = false;
 		strncpy(de.d_name, ".", 1);
```

Comment:
shouldn't these `strncpy`s just be an assignment now?

Reply:
I did not want to touch that a lot (yet I noticed my renaming introduced a mistake ...).

Ok but I've found a ctor of std::string that ought to supersede the 2nd `strncpy`.

Reply:
Soft-tested my changes on godbolt